modules/number-picker/main/index.coffee

```diff
@@ -72,27 +72,33 @@ class NumberPicker extends hx.EventEmitter
     if @options.max isnt undefined then @max @options.max
     if @options.min isnt undefined then @min @options.min
     if @options.disabled then @disabled(@options.disabled)
-
-    @selectInput
-      .attr('data-value', @options.value)
-      .value(@options.value)
+    @value @options.value
 
   value: (value, screenValue) ->
     if arguments.length > 0
       prevValue = @value()
-      newVal = if not value? and screenValue then screenValue else value
 
-      if @_.max isnt undefined and newVal > @_.max then newVal = @_.max
-      if @_.min isnt undefined and newVal < @_.min then newVal = @_.min
+      valueToUse = if not value? and screenValue then Number(screenValue) else value
+
+      newVal = checkValue(valueToUse, @min(), @max())
```

Comment:
It would seem that hx.clamp doesn't handle undefined for min or max at the moment, so hx.clamp would need an upgrade to be used here